drivers/pinctrl/pinctrl_sy1xx.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 sensry.io
+ */
+
+#define DT_DRV_COMPAT sensry_sy1xx_pinctrl
+
+#include <zephyr/drivers/pinctrl.h>
+
+#include <zephyr/arch/common/sys_io.h>
+
+#define SY1XX_PAD_CONFIG_BASE_ADDR      (0x1A104000)
+#define SY1XX_PAD_CONFIG_BASE_ADDR_MASK (0xFFFFF000)
+
+#define FIELD_REPLACE(mask, value, field_value)                                                    \
+	(((value) & ~(mask)) | FIELD_PREP(mask, field_value))
```

Comment:
I guess this could likely be put together with the FIELD_GET/PREP

Reply:
actually I tried. but didnt succeed.

given a `0x11223344` I need to replace ex: `22` by something like `a5`. or replace `33` by something like `82`. I couldnt find a way to use only `FIELD_PREP` and `FIELD_GET`

Reply:
suggest this to work:

```
	uint32_t reg = ~(0xFFUL << pin->iro) & sys_read32(pin->addr);
	reg |= FIELD_PREP((0xFFUL << pin->iro), pin->cfg);
	sys_write32(reg, pin->addr);

```

Reply:
solved, from my perspective.